llvm/lib/CASObjectFormats/NestedV1.cpp

```diff
@@ -624,7 +624,7 @@ static bool compareSymbolsBySemanticsAnd(
     function_ref<bool(const jitlink::Symbol *, const jitlink::Symbol *)>
         NextCompare) {
   if (LHS == RHS)
-    return false;
+    return NextCompare(LHS, RHS);
```

Comment:
Haven't had a chance to think through the rest yet, but can you explain why this is necessary? If the symbols are equal, why compare further? (Can/should it be split to a different commit since it seems like a functionality change?)

Reply:
Ah, I see the call in `visitEdgesInStableOrder()`, I think I understand.

Reply:
> Ah, I see the call in visitEdgesInStableOrder()

Right, I also checked the other uses of that function and it doesn't change existing functionality, AFAICT.